src/components/Modals/UploadButtonGroup.jsx

```diff
@@ -19,22 +20,70 @@ import { useTheme } from '@mui/material/styles';
  * @param {Function} Props.setFile - The set function for handling files
  * @returns {React.JSX.Element} - The UploadButtonGroup Component
  */
+
+const checkCameraAvailability = async () => {
+  try {
+    const devices = await navigator.mediaDevices.enumerateDevices();
+    const videoInputDevices = devices.filter((device) => device.kind === 'videoinput');
+    return videoInputDevices.length > 0;
+  } catch (error) {
+    // console.error("Error checking camera availability: ", error);
+    return false;
+  }
+};
+
+const dataURItoBlob = (dataURI) => {
+  // Convert base64/URLEncoded data component to raw binary data held in a string
+  let byteString;
+  if (dataURI.split(',')[0].indexOf('base64') >= 0) byteString = atob(dataURI.split(',')[1]);
+  else byteString = unescape(dataURI.split(',')[1]);
+
+  // Separate the mime component
+  const mimeString = dataURI.split(',')[0].split(':')[1].split(';')[0];
+
+  // Write the bytes of the string to a typed array
+  const ia = new Uint8Array(byteString.length);
+  for (let i = 0; i < byteString.length; i += 1) {
+    ia[i] = byteString.charCodeAt(i);
+  }
+
+  return new Blob([ia], { type: mimeString });
+};
+
 const UploadButtonGroup = ({ file, setFile }) => {
-  const theme = useTheme();
-  const isSmallScreen = useMediaQuery(theme.breakpoints.down('sm'));
-  const isCustomMediumScreen = useMediaQuery('(max-width: 768px)');
+  const [hasCamera, setHasCamera] = useState(false);
```

Comment:
Rather than using `hasCamera`/`setHasCamera` for the state, it might be better to use `hasWebCam`/`setHasWebCam`.

This makes things less ambiguous since `hasCamera` could refer to any camera on the device. Having `hasWebCam` directly refers to a specific camera on the devices which the earlier function targets.